src/wazuh_modules/main.c

```diff
@@ -80,8 +80,10 @@ int main(int argc, char **argv)
     mdebug1(WAZUH_HOMEDIR, home_path);
     os_free(home_path);
 
-    // Setup daemon
+    // Signal management initialization must be before wm_setup()
+    wm_signals_configure();
```

Comment:
Good catch, it's true that the configure flag was leaving a dangling PID file.
But this re-introduces the problem of #26536, you can try with this method to confirm
https://github.com/wazuh/wazuh/issues/26536#issuecomment-2437564978

The signal manipulation was set after the module start on purpose to allow the module to completely setup before handling any stop that could arrive to early.

See https://github.com/wazuh/wazuh/pull/26679

>The signals are ignored at startup and until the modules have started. This avoids the termination of the module during the sleep() calls.

I suggest to split `wm_signals_configure()`, leave the signal handling setup where it is and move up the `atexit(wm_cleanup);` only.

Reply:
It makes sense, thanks